2-js-basics/2-functions-methods/translations/README.fr.md

```diff
@@ -0,0 +1,200 @@
+
+[![Méthodes et fonctions](https://img.youtube.com/vi/XgKsD6Zwvlc/0.jpg)](https://youtube.com/watch?v=XgKsD6Zwvlc "Méthodes et fonctions")
+
+> 🎥 Cliquez sur l'image ci-dessus pour une vidéo sur les méthodes et les fonctions.
+
+> Vous pouvez suivre cette leçon sur [Microsoft Learn](https://docs.microsoft.com/en-us/learn/modules/web-development-101-functions/?WT.mc_id=academic-13441-cxa) !
+
+## Les fonctions
+
+À la base, une fonction est un bloc de code que nous pouvons exécuter à la demande. C'est parfait pour les scénarios où nous devons effectuer la même tâche plusieurs fois ; plutôt que de dupliquer la logique à plusieurs endroits (ce qui rendrait difficile la mise à jour le moment venu), nous pouvons la centraliser à un seul endroit et l'appeler chaque fois que nous avons besoin de l'opération effectuée - vous pouvez même appeler des fonctions à partir d'autres fonctions !.
+
+La capacité de nommer une fonction est tout aussi importante. Bien que cela puisse sembler trivial, le nom fournit un moyen rapide de documenter une section de code. Vous pourriez penser à cela comme une étiquette sur un bouton. Si je clique sur un bouton indiquant « Annuler la minuterie », je sais que l'horloge va s'arrêter.
+
+## Créer et appeler une fonction
+
+La syntaxe d'une fonction ressemble à la suivante :
+
+```javascript
+function nameOfFunction() { // function definition
+ // function definition/body
+}
+```
+
+Si je voulais créer une fonction pour afficher un message d'accueil, cela pourrait ressembler à ceci :
+
+```javascript
+function displayGreeting() {
+  console.log('¡Hola, mundo!');
+}
+```
+
+Chaque fois que nous voulons appeler (ou invoquer) notre fonction, nous utilisons le nom de la fonction suivi de `()`. Il convient de noter que notre fonction peut être définie avant ou après que nous décidions de l'appeler ; le compilateur JavaScript le trouvera pour vous.
+
+```javascript
+// calling our function
+displayGreeting();
+```
+
+> **REMARQUE :** Il existe un type spécial de fonction connu sous le nom de **méthode**, que vous avez déjà utilisé ! En fait, nous l'avons vu dans notre démo ci-dessus lorsque nous avons utilisé `console.log`. Ce qui différencie une méthode d'une fonction, c'est qu'une méthode est attachée à un objet (`console` dans notre exemple), tandis qu'une fonction est flottante. Vous entendrez de nombreux développeurs utiliser ces termes de manière interchangeable.
+
+### Bonnes pratiques de la fonction
+
+Il y a une poignée de bonnes pratiques à garder à l'esprit lors de la création de fonctions
+
+- Comme toujours, utilisez des noms descriptifs pour savoir ce que la fonction fera
+- Utilisez **boîte de chameau** pour combiner des mots
```

Comment:
`boîte de chameau` doesn't have sense here from my point of vue, the `camelCase` word is also used in french and `boîte de chameau` is not appropriated for the context i think